### PR TITLE
first pass at simple chemical query

### PR DESCRIPTION
**Pull request by kjaym**
@maxhutch This adds the `simple_chemical_search` method that lets us generate quick, common use case queries without having to build up entire PifQuery objects.

### CODE REVIEW

File: citrination_client/client.py [hunk 1]
```diff
@@ -55,6 +65,116 @@ def search(self, pif_query):
...
+            system_query.chemical_formula = \
+                [ChemicalFieldOperation(extract_as='chemical_formula', filter=ChemicalFilter(equal=chemical_formula))]
+        else:
+            system_query.chemical_formula = \
+                [ChemicalFieldOperation(extract_as='chemical_formula')]
+        
+        # Generate the name query for the property
+        if isinstance(property_name, list):
+            property_name_query = \
+                [FieldOperation(extract_as='property_name', filter=Filter(equal=i)) for i in property_name]
```
Comment (maxhutch) [inline citrination_client/client.py]: Shouldn't this be:
```
property_name_query = FieldOperation(extract_as='property_name', filter=[Filter(equal=i) for i in property_name])
```

File: citrination_client/client.py [hunk 2]
```diff
@@ -55,6 +65,116 @@ def search(self, pif_query):
...
+        """
+        system_query = SystemQuery()
+
+        # Add the name operation
+        if isinstance(name, list):
+            system_query.names = [FieldOperation(extract_as='name', filter=Filter(equal=i)) for i in name]
+        elif isinstance(name, string_types):
+            system_query.names = [FieldOperation(extract_as='name', filter=Filter(equal=name))]
+        else:
+            system_query.names = [FieldOperation(extract_as='name')]
```
Comment (sinistral) [inline citrination_client/client.py]: This pattern of 
```
if list, then then return list of reshaped items
else if string then return list of reshaped strings
else return reshaped thing
```
seems to repeat through the rest of the this method.  Extracting it as a function would reduce both the duplication and the obfuscation of the high-level "recipe" that results from the `if/elif/else` blocks.

File: citrination_client/client.py (hunk 1 above)
Comment (kjaym) [inline citrination_client/client.py]: Both of those work just fine, actually. It looks like that page in the documentation is off. The property name can accept a list of FieldOperations (https://github.com/CitrineInformatics/python-citrination-client/blob/master/citrination_client/search/pif/query/core/property_query.py#L19). 

However, now that you mention it, even though both queries are allowed, the one that you wrote would be more efficient to execute! I am going to rewrite these blocks.

**Comment by kjaym** [comment]
@sinistral This method is really just a builder for the underlying queries that we use. It does not map directly to any parts of the API. Having a high level API though would be a nice idea if this resonates with people! It would be easy to swap out the code in the method with a call to that _future_ API.

File: citrination_client/client.py (hunk 1 above)
Comment (maxhutch) [inline citrination_client/client.py]: It wasn't working for me.  When I passed in two property names (for the same property), I got no results back.  When I removed either of the names, to get down to a list with a single member, I got results again.

**Comment by sinistral** [comment]
Is there a way to track which of the client methods are actually being used (e.g. a header) so that we can measure which pieces of high-level functionality are seeing use relative to the underlying building blocks?

**Comment by kjaym** [comment]
That is a very good idea, but one we are not doing now!

File: citrination_client/client.py (hunk 1 above)
Comment (kjaym) [inline citrination_client/client.py]: Okay, I know exactly what you were seeing. We can talk about that later, but it is currently intended behavior. Either way, the changes I am making will negate that issue for this PR.

**Comment by kjaym** [comment]
@maxhutch @sinistral Just pushed code that simplifies the query building logic.

File: citrination_client/client.py [hunk 3]
```diff
@@ -55,6 +63,94 @@ def search(self, pif_query):
...
+        """
+        Helper method that wraps values in a list. If the input is a list then it is returned. If the input is None
+        then an empty list is returned. For anything else, the input value is wrapped as a single-element list.
+
+        :param values: Value to make sure exists in a list.
+        :return: List with the input values.
+        """
+        if values is None:
+            return []
+        elif isinstance(values, list):
```
Comment (sinistral) [inline citrination_client/client.py]: Is a `tuple` also be a legal input sequence, i.e. `simple_chemical_search(name=('A', 'B'))`

**Comment by sinistral** [comment]
Simplified logic is simple.

File: citrination_client/client.py (hunk 3 above)
Comment (kjaym) [inline citrination_client/client.py]: Not at the moment. That is consistent with the rest of this and our related python tools though.

**Comment by maxhutch** [comment]
LGTM

**Comment by kjaym** [comment]
@sinistral Any other comments?

**Pull request merged**